native/Makefile

```diff
@@ -289,10 +291,11 @@ LDFLAGS+=-Wl,-soname,$@,-Bsymbolic
 endif
 endif
 
-ifneq (,$(findstring bsd,$(OS)))
+ifneq (,$(or $(findstring bsd,$(OS)),$(findstring dragonfly,$(OS))))
 ARCH=$(shell uname -m | sed 's/i.86/i386/g')
 PCFLAGS+=-fPIC
-CINCLUDES+=-I/usr/X11R6/include
+X11INCDIR?=/usr/X11R6/include
```

Comment:
Is this really needed? You describe, that it might be required on FreeBSD, but I checked the build-package I use to generate the freebsd package and it just works without this modification.

Reply:
I just tried it on FreeBSD and this change is indeed not needed on it; however, it was necessary on DragonFly. This was interesting, so I tracked it down.

As it turned out, the default search paths of `gcc` were different:

- On FreeBSD, the `gcc` is installed via `pkg(8)` and resides in `/usr/local/bin/gcc`, so the default search paths contain `/usr/local/include`.

    ```
    $ which gcc
    /usr/local/bin/gcc

    $ gcc -xc -E -Wp,-v /dev/null
    #include "..." search starts here:
    #include <...> search starts here:
     /usr/local/lib/gcc12/gcc/x86_64-portbld-freebsd13.2/12.2.0/include
     /usr/local/include
     /usr/include
    End of search list.
    ```

- On DragonFly, the (default) `gcc` is shipped with the OS and resides in `/usr/bin/gcc`, so the default search paths do **not** contain `/usr/local/include`.

    ```
    $ which gcc
    /usr/bin/gcc

    $ gcc -xc -E -Wp,-v /dev/null
    #include "..." search starts here:
    #include <...> search starts here:
     /usr/libdata/gcc80
     /usr/include
    End of search list.
    ```

Reply:
I'd like we make this change so that we're more independent of the specific compilers.

What do you suggest? Thanks.

Reply:
I can live with this as is, but we could also add `/usr/local/include` always, then the build would work on all BSDs identically out of the box. What do you think?

Reply:
I just checked NetBSD (9.1) and OpenBSD (7.4) and this is a mess.  Both NetBSD and OpenBSD bundles X11 in their OS; however, the X11 headers locate in `/usr/X11R7/include` on NetBSD but in `/usr/X11R6/include` on OpenBSD...

Therefore, I'd like to make this segment to be following, avoiding multiple `ifeq`s.

```makefile
ifneq (,$(or $(findstring bsd,$(OS)),$(findstring dragonfly,$(OS))))
ARCH=$(shell uname -m | sed 's/i.86/i386/g')
PCFLAGS+=-fPIC
# This is a mess: X11 headers locate in /usr/local/include on FreeBSD and DragonFly,
# in /usr/X11R7/include on NetBSD, and in /usr/X11R6/include on OpenBSD.
CINCLUDES+=-I/usr/local/include -I/usr/X11R7/include -I/usr/X11R6/include
CINCLUDES+=$(X11INC)  # allow extra X11 include path if necessary
LDFLAGS=-o $@ -shared
CDEFINES+=-DHAVE_PROTECTION -DFFI_MMAP_EXEC_WRIT -DUSE_DEAFULT_LIBNAME_ENCODING
endif
```